src/operator/tensor/broadcast_reduce_op_value.cc

```diff
@@ -138,7 +138,16 @@ NNVM_REGISTER_OP(broadcast_like)
     [](const NodeAttrs& attrs) {
       return std::vector<std::string>{"lhs", "rhs"};
     })
-.set_attr<nnvm::FInferType>("FInferType", ElemwiseType<2, 1>)
+.set_attr<nnvm::FInferType>("FInferType", [](const nnvm::NodeAttrs& attrs,
+                                             std::vector<int> *in_attrs,
+                                             std::vector<int> *out_attrs) {
+  CHECK_EQ(in_attrs->size(), 2) << " in operator " << attrs.name;
+  std::vector<int> checked_in_attrs = { (*in_attrs)[0] };
+  bool ret = !type_is_none((*in_attrs)[1]) &&
```

Comment:
Thanks for your contribution!

Is it necessary for the condition `!type_is_none((*in_attrs)[1])` ?

Reply:
Well, I actully just found the issue,  the code is copied from merged PR #14097, which had a similar issue. I think it is better to keep the same requirement for all *_like op.

Reply:
Yes, it is necessary - what `FInferType` returns is whether it succeeded in inferring all the types (so that if all operators return true we know that all types are inferred). That is why it is important to not lie and return true only if all types are really inferred (even if we do not actually do anything with the other type).